crates/aggchain-proof-service/src/service/customchaindata_builder.rs

```diff
@@ -0,0 +1,281 @@
+use std::{
```

Comment:
Move this service to the separate crate, e.g. `aggchain-custom-chain-data-builder` . 
Add Error and Config types for it. 
It should be consistent with the other services.